framework/include/interfaces/Coupleable.h

```diff
@@ -135,6 +135,16 @@ class Coupleable
    */
   virtual const VariableValue & coupledValue(const std::string & var_name, unsigned int comp = 0);
 
+  /**
+   * Returns value of a coupled variable for use in templated auatomatic differentiation classes
+   * @param var_name Name of coupled variable
+   * @param comp Component number for vector of coupled variables
+   * @return Reference to a VariableValue for the coupled variable
+   */
+  template <bool is_ad = false>
```

Comment:
I advise against adding a defaults here. You got bitten by that recently. If you forget the explicit template argument you will get a behavior this function is not designed for.

Reply:
Not sure if we have any reason at all to have a default `is_ad` anywhere in the code. Most likely I got it from `FunctionMaterialBase`